lib/ddtrace/propagation/distributed_headers.rb

```diff
@@ -11,8 +11,10 @@ def initialize(env)
     end
 
     def valid?
-      # Sampling priority and origin are optional.
-      trace_id && parent_id
+      # We just need a valid trace id. Parent id, sampling priority, and origin are all optional
+      # DEV: No parent id means that our trace will be the parent, but we use a specific trace id
+      #      this is the case for Synthetics requests, they send `X-Datadog-Parent-Id: 0`
+      trace_id
```

Comment:
Synthetics aside, if distributed tracing were configured and service A were misconfigured such that it only set `Trace ID` and not `Parent ID`, then when both service A and service B submit their traces with the same `Trace ID` and `Parent ID = 0`, won't that be considered a collision?

Reply:
We don't allow a value of `0` for either `trace_id` or `parent_id`, so here it'll end up being `nil`.

So, yes, could still potentially cause an issue, but this logic is similar to what other tracers do, and fixes issue for Synthetics :/

Reply:
The issue is two traces having the same trace id but not being linked by a parent id.

Reply:
In that case there will be a collision, but the backend will ingest both regularly. The UI will show just the first one (so the "root trace") and hence this is not supported. 

The concern makes sense. Having the `parent_id == 0` is something valid (synthetics or not synthetics) because it's a correct representation of a tracing context. But in the sense of distributed tracing it is not because we're not expecting that a tracer sends `0` as a parent. We can consider this an edge case, but I'd like to be defensive in the "unhappy path".

What if:
* we keep this condition that enforces `trace_id` and `parent_id` to not be `0` (happy path -> distributed tracing between services)
* we add a check if this is a synthetic trace (i.e. checking the `origin` header) and if it's a synthetic trace we only check `trace_id` (because in the case of synthetics this is legit). It covers the happy path (synthetics) and every unhappy path for distributed tracing